### ISSUE TITLE
Timeseries example does NOT address the core of prediction

### ISSUE THREAD

**Issue by summa-code** [tensorflow/tensorflow#45970]
## URL(s) with the issue:
https://www.tensorflow.org/tutorials/structured_data/time_series

Please provide a link to the documentation entry, for example:
https://www.tensorflow.org/tutorials/structured_data/time_series

## Description of issue (what needs changing):
I get the sequential data and the methods to get to next step or multi steps. But with the Time being converted to frequency, why do we have to have again equidistant events record. That beats the purpose of ANY timed series problem. In the example given, converting the time to frequency does not affect the model much because the series is Timed, not any value for the period. The period between records is equal. I still don't see this as a clear example of Time series. You can call this example as Series prediction, not Timed series prediction.

1) There seems to be some errors too,  In the below function, why do we have Shuffle to TRUE ? I thought timeseries is sequential ? 

```
def make_dataset(self, data):
  data = np.array(data, dtype=np.float32)
  ds = tf.keras.preprocessing.timeseries_dataset_from_array(
      data=data,
      targets=None,
      sequence_length=self.total_window_size,
      sequence_stride=1,
      shuffle=True,
      batch_size=32,)
  ds = ds.map(self.split_window)
  return ds
```

2 ) And also, how is the Validation done ? Do we have to window slide the validation data too ?
3)  Do we really need window slide for LSTM ? Seems to be over fitting for a lots of examples.
4)  If some input fields have categorical data (not the output), there is no way this will perform better especially if we use embedding
5) With this function timeseries_dataset_from_array, it is hard to do any embedding layer.
6) Timeseries involves large datasets, unfortunately using BatchNormalization is very confusing, blame it on the documentation.
7) How do we do feature engineered layer (such as mixed with embedding layer) in the first input with 3D tensor.

**Comment by summa-code** [comment]
And also i don't understand this one in the code in make dataset function

       ```data = np.array(data, dtype=np.float32)```
        ds = tf.keras.preprocessing.timeseries_dataset_from_array(
            data=data,
            targets=None,
            sequence_length=self.total_window_size,
            sequence_stride=sequence_stride,
            shuffle=shuffle,
            batch_size=batch_size, )
If we have to preserve the dtype from dataframe until we feed this to feature engineered layer, how do we do it ?

**Comment by summa-code** [comment]
1) There seems to be some errors too,  In the below function, why do we have Shuffle to TRUE ? I thought timeseries is sequential ? 

```
def make_dataset(self, data):
  data = np.array(data, dtype=np.float32)
  ds = tf.keras.preprocessing.timeseries_dataset_from_array(
      data=data,
      targets=None,
      sequence_length=self.total_window_size,
      sequence_stride=1,
      shuffle=True,
      batch_size=32,)
  ds = ds.map(self.split_window)
  return ds
```

2 ) And also, how do the Validation done ? Do we have to window slide the validation data too ?
3)  Do we really need window slide for LSTM ? Seems to be over fitting for a lots of examples.
4)  If some input fields have categorical data (not the output), there is no way this will perform better especially if we use embedding
5) Timeseries involves large datasets, unfortunately using BatchNormalization is very confusing, blame it on the documentation.

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
